homeassistant/components/octoprint/__init__.py

```diff
@@ -181,10 +181,11 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     session = aiohttp.ClientSession(connector=connector)
 
     @callback
-    def _async_close_websession(event: Event) -> None:
+    def _async_close_websession(event: Event | None = None) -> None:
         """Close websession."""
         session.detach()
 
+    entry.async_on_unload(_async_close_websession)
     hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, _async_close_websession)
```

Comment:
We should change this to use `async_listen` and wrap it with `entry.async_on_unload` to avoid leaking the listener when the config entry is later unloaded.

Using `async_listen` is important here because the entry could be unloaded after the stop event, which might otherwise lead to the listener being canceled twice.

Reply:
Applied :)

Thanks for the review!